### ISSUE TITLE
installation problem on Ubuntu 18.04

### ISSUE THREAD

**Issue by muralikodali** [bilde2910/Hauk#12]
after manual installation of memcached server , when i tried to access hauk page on my apache server , it displayed location expired page.

**Comment by bilde2910** [comment]
Currently, visiting the root page (https://hauk.example.com/) will always result in "Location expired." A fix is planned for this soon. You need a valid session URL (https://hauk.example.com/?ABCD-1234) to actually view the location - launch the Hauk app, add your Apache server URL as the server URL in the app, start sharing, then you can copy the link either using the "Share" button, or copying the link from Hauk's notification. Can you confirm if this works?

**Comment by muralikodali** [comment]
Followed your suggestion.
I have got different responses (http and https - selfsigned) , when i tried to access the server from hauk android app.

screenshots attached for your reference.

![https_selfsigned_error](https://user-images.githubusercontent.com/11482766/64407713-a8fdde80-d0a2-11e9-9acf-f2fd025628e9.png)
![http_error](https://user-images.githubusercontent.com/11482766/64407714-a8fdde80-d0a2-11e9-9aeb-af7368829637.png)

**Comment by bilde2910** [comment]
The HTTPS error is because of your self signed certificate root not being trusted by your phone. You *could* add the certificate to your phone's trusted certificates store, but that is not a secure solution. I would rather strongly encourage you to use Let's Encrypt certificates for your server, as these are free to deploy and trusted by Android. You can find information on setting up Let's Encrypt on [EFF's certbot website](https://certbot.eff.org/).

As for the HTTP response, can you verify that you are entering the path to the Hauk installation folder itself (e.g. https://hauk.example.com/) and **not** the index.html file contained within it (https://hauk.example.com/index.html)?

**Comment by muralikodali** [comment]
As for HTTP response, I have entered URL without index.html.
As for HTTPS , I request you to add option for trusting self-signed certificate , which is a convenient way for servers that are dependent upon dynamic dns without FQDN.

**Comment by RomanJos** [comment]
You need to import your cert into your phone settings https://stackoverflow.com/questions/37281958/how-to-trust-self-signed-certificate-on-android

**Comment by bilde2910** [comment]
Allowing self-signed certificates completely eliminates the security benefit of using HTTPS in the first place, so this will not be done. Even if the self-signed certificate is added to your phone's trust store with the method @RomanJos linked to, the recipient of your shares will not have a way to trust and verify your self-signed certificate on their end; anyone could intercept the traffic and present their own self-signed cert instead, and the recipient would be none the wiser. Furthermore, installing your own root CA to your device is extremely dangerous.

The only situation in which you cannot use Let's Encrypt for HTTPS is if you try to get a certificate for an IP address. Everything else will work. Using a dynamic DNS service will not stop you from getting a certificate from Let's Encrypt (see the "webroot" verification method).

---

Regarding the HTTP issue, it looks like your server is misconfigured in that case. It appears to be redirecting all traffic to the index page, even traffic to other files/subdirectories. I need your server config to determine what's wrong. In particular, I'll need the section of your configuration that concerns your vhost/domain, e.g. the `server {}` block in nginx or `<VirtualHost />` in Apache, if you are willing to share those.

**Comment by muralikodali** [comment]
Thanks for your guidance on installing letsencrypt with webroot verification method. I have successfully installed letsencrypt certificate.
Even after installing letsencrypt, i am getting the same response i have previously got from http://  url. I am attaching my apache config files as suggested by you.
[apacheconf.tar.gz](https://github.com/bilde2910/Hauk/files/3587730/apacheconf.tar.gz)

**Comment by bilde2910** [comment]
I'm totally stuck on this. Your config looks fine to me, but I'm sure I'm missing something. Do you have any other PHP scripts running, and if so, do they work?

As a last resort, can you use the Docker image and use Apache or another httpd as a reverse proxy for it?

**Comment by muralikodali** [comment]
Now i am getting a sharing link. But status is shown: **Waiting for GNSS fix..."** on my mobile app.
I am able to share my location on google maps.

**Comment by bilde2910** [comment]
Are location services set to "high accuracy" on your device? Also, are you running the latest version of the app? I pushed an update a few days ago that improves location fixes. The "waiting for initial GNSS fix" status appears when your device can't get a GPS signal.

**Comment by muralikodali** [comment]
In android 9 pie , there is no option to set it  to high accuracy. I am running the latest  version of the app.
In GPS Status app: when location source is set to google , I am able to get GPS location. When it is set to GPS (device only), location is not displayed. May be some problem with device GPS.

**Comment by bilde2910** [comment]
Probably the device then, but this should have been fixed in #11. Not entirely sure why this fix doesn't work for you. Maybe I've missed something, but I'm really not sure where to go next.

**Comment by muralikodali** [comment]
I have updated hauk on Android to **V1.1.1** and it is **now working well** with status showing "**waiting for high accouracy fix**..."
Now the problem is accuracy, showing straight lines instead of all curves and bends,                                                                                                                                                                                                                                                                      which is dependent upon device GPS. It is not the problem concerned with hauk.

**Comment by bilde2910** [comment]
I realized that v1.1.1 wasn't actually released on F-Droid yet when I wrote the last comment, but looks like you figured it out :-)

The straight lines issue sounds like a device-specific problem. Usually means that the device is only using network-based location instead of GPS constellations. Normally most devices will switch over to GPS based location whenever it becomes available, but again that depends on the device, so you might indeed have more luck looking up the cause of that elsewhere.
